include/lbann/utils/random_number_generators.hpp

```diff
@@ -29,12 +29,21 @@
 
 #include "lbann/comm.hpp"
 #include <random>
+#include <mutex>
+#include <thread>
 
 namespace lbann {
 
 using rng_gen = std::mt19937;  // Mersenne Twister
 using fast_rng_gen = std::minstd_rand;  // Minimum standard, LC
 
+struct io_rng_t {
+  lbann::rng_gen io_generator;
+  lbann::fast_rng_gen fast_io_generator;
+  std::unique_ptr<std::mutex> io_mutex;
```

Comment:
Pedantic style comment: the `io` in the variable names is redundant with the name of the struct `io_rng_t`.